jmx/src/main/java/io/airlift/jmx/JavaVersion.java

```diff
@@ -35,7 +35,10 @@
     private static final Pattern PATTERN = Pattern.compile(VERSION_NUMBER + PRE + BUILD + OPT);
 
     // For Java 8 and below
-    private static final Pattern LEGACY_PATTERN = Pattern.compile("1\\.(?<MAJOR>[0-9]+)(\\.(?<MINOR>(0|[1-9][0-9]*)))?(_(?<UPDATE>[1-9][0-9]*))?(?:-ea)?");
+    private static final String LEGACY_REGEX = "1\\.(?<MAJOR>[0-9]+)(\\.(?<MINOR>(0|[1-9][0-9]*)))?(_(?<UPDATE>[1-9][0-9]*))?(?:-ea)?";
```

Comment:
This seems overkill. We don't need two matchers (LEGACY_PATTERN vs LEGACY_CUSTOM_PATTERN). All we need to do is extend the pattern for -ea to match any combination of letters and numbers.